### PR TITLE
Adding HPC logging and extending logger to support different WG

### PR DESCRIPTION
**Pull request by azrael417**
Dear Sir/Madam,

I have extended the existing package checker to be able to support other WG's such as HPC (only HPC and training are supported yet, where HPC is WIP). This PR does not touch the evaluation or checking logic but makes use of the already existing usage parameter and extends the compliance checker and other relevant scripts such as the rcp checker with that usage parameter. Mostly what was changed was renaming the directories named <ruleset> into <usage>_<ruleset>. That way, the checker can pick up the right compliance and rcp scripts from the correct WG.

I am happy to discuss the changes here but I open the PR to get things started. HPC submission deadline is end of September and we would like to have a working submission checker stack by then.

Note that if the usage parameter is not passed, it defaults to training so it should in principle not break any BW compatibility. I tested the script on the NVIDIA training submission for 1.0.0 and it passed, but feel free to verify on your own in case I missed subtle details. 

Some things which could be improved:

- the dictionaries containing the allowed benchmarks and the required number of runs are in the python scripts, but ideally those should be read from a YAML file as well. I haven't done that yet because I wanted to keep the changes minimally invasive, but I am happy to do that.
- HPC scoring has an additional metric which I am not yet clear about of how to incorporate into this stack and thus it is missing yet. I will think about it a bit and will open another PR for this later on.

Let me know what you think

Best and thanks
Thorsten

### CODE REVIEW

**Comment by emizan76** [comment]
Hi Thorsten,

Thank you for your contribution and comments.
Adding HPC to the mainline training logging mostly makes sense, but also
comes as a surprise.
At least from my side I would like to know more of HPC training plans, what
you expect from the submission infra, if and for how long you can
contribute, etc.

Can you please attend our weekly meeting one time, on Tuesday morning
10:30PDT?
What is the best e-mail address so we send you the invitation?

Thanks,

Elias

**Comment by azrael417** [comment]
Hello Elias,

I am happy to join. Can you send me an invitation, maybe for tomorrow?

Btw, I did a rebase to fix the author name + email issue. My email is tkurth@nvidia.com.

Best regards
Thorsten

File: mlperf_logging/compliance_checker/hpc_1.0.0/common.yaml
Comment (sparticlesteve) [inline mlperf_logging/compliance_checker/hpc_1.0.0/common.yaml:148]: we use eval_error for the other two benchmarks, so this has to be removed from common

File: mlperf_logging/compliance_checker/hpc_1.0.0/closed_common.yaml
Comment (emizan76) [inline mlperf_logging/compliance_checker/hpc_1.0.0/closed_common.yaml:5]: Remove on deepcam

File: mlperf_logging/package_checker/package_checker.py
Comment (emizan76) [inline mlperf_logging/package_checker/package_checker.py:265]: Is hpc_1.0.0 really accepted here? I think hpc is part of usage

File: mlperf_logging/compliance_checker/hpc_1.0.0/closed_common.yaml
Comment (emizan76) [inline mlperf_logging/compliance_checker/hpc_1.0.0/closed_common.yaml:5]: Remove one deepcam

File: mlperf_logging/compliance_checker/mlp_compliance.py
Comment (emizan76) [inline mlperf_logging/compliance_checker/mlp_compliance.py:44]: Have you tried to run standalone MLP compliance for a single log file? I tried and it fails, if you have please provide the command line for hpc/training. Also probably you will need to change the compliance_checker/README.md dile

File: mlperf_logging/compliance_checker/mlp_parser/__init__.py
Comment (emizan76) [inline mlperf_logging/compliance_checker/mlp_parser/__init__.py:7]: I think we may need separate ruleset files for hpc and training.

**Review by emizan76 — COMMENTED** [review]
I made a few comments -- looks good for the most part.

File: mlperf_logging/package_checker/package_checker.py
Comment (azrael417) [reply mlperf_logging/package_checker/package_checker.py:265]: You are right, I fixed this.

File: mlperf_logging/compliance_checker/hpc_1.0.0/closed_common.yaml
Comment (azrael417) [reply mlperf_logging/compliance_checker/hpc_1.0.0/closed_common.yaml]: fixed.

File: mlperf_logging/compliance_checker/mlp_compliance.py
Comment (azrael417) [reply mlperf_logging/compliance_checker/mlp_compliance.py:44]: I ran

`python3 -m mlperf_logging.package_checker <1.0.0 results folder> training 1.0.0 --rcp_bert_train_samples`

on the NVIDIA and Google folders and everything passed. For example for the Google result:

```SUCCESS
Table CSV prefix: Google,tpu-v4-256,AMD Rome,256,TPU,4,JAX,
Table CSV postfix: ,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/systems/tpu-v4-256.json,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/benchmarks,
SUCCESS
Table CSV prefix: Google,tpu-v4-4096,AMD Rome,256,TPU,4,TensorFlow,
Table CSV postfix: ,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/systems/tpu-v4-4096.json,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/benchmarks,
SUCCESS
Table CSV prefix: Google,tpu-v4-2048,AMD Rome,256,TPU,4,TensorFlow,
Table CSV postfix: ,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/systems/tpu-v4-2048.json,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/benchmarks,
SUCCESS
Table CSV prefix: Google,tpu-v4-256,AMD Rome,256,TPU,4,TensorFlow,
Table CSV postfix: ,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/systems/tpu-v4-256.json,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/benchmarks,
SUCCESS
Table CSV prefix: Google,tpu-v4-128,AMD Rome,256,TPU,4,TensorFlow,
Table CSV postfix: ,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/systems/tpu-v4-128.json,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/benchmarks,
SUCCESS
Table CSV prefix: Google,tpu-v4-1024,AMD Rome,256,TPU,4,JAX,
Table CSV postfix: ,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/systems/tpu-v4-1024.json,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/benchmarks,
SUCCESS
Table CSV prefix: Google,tpu-v4-6912,AMD Rome,256,TPU,4,TensorFlow,
Table CSV postfix: ,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/systems/tpu-v4-6912.json,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/benchmarks,
SUCCESS
Table CSV prefix: Google,tpu-v4-128,AMD Rome,256,TPU,4,JAX,
Table CSV postfix: ,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/systems/tpu-v4-128.json,https://github.com/mlperf/training_results_v1.0.0/blob/master/Google/benchmarks,```

File: mlperf_logging/compliance_checker/mlp_parser/__init__.py
Comment (azrael417) [reply mlperf_logging/compliance_checker/mlp_parser/__init__.py]: I have fixed this, this was actually not necessary.

File: mlperf_logging/package_checker/package_checker.py
Comment (azrael417) [reply mlperf_logging/package_checker/package_checker.py:95]: I think we can change it to `check_result_files`, what do you think?

File: mlperf_logging/package_checker/package_checker.py
Comment (azrael417) [reply mlperf_logging/package_checker/package_checker.py:104]: I think this is a matter of style. I checked some other asserts on the code and they use parenthesis. I wanted to keep that style. Shall I remove those?

File: mlperf_logging/package_checker/package_checker.py
Comment (azrael417) [reply mlperf_logging/package_checker/package_checker.py]: I have changed this in the spirit you mentioned. I did the same in result summarizer and moved all the benchmark file counts and allowed benchmark dictionaries into a file called benchmark_meta.py. This is used by result summarizer and package checker, so that once more benchmarks are added, just a single file needs to be changed. I hope you like that refactor.

File: mlperf_logging/package_checker/package_checker.py
Comment (azrael417) [reply mlperf_logging/package_checker/package_checker.py]: fixed

File: mlperf_logging/result_summarizer/result_summarizer.py
Comment (azrael417) [reply mlperf_logging/result_summarizer/result_summarizer.py:306]: fixed

File: mlperf_logging/result_summarizer/result_summarizer.py
Comment (azrael417) [reply mlperf_logging/result_summarizer/result_summarizer.py]: Done

Comment (azrael417) [reply mlperf_logging/result_summarizer/result_summarizer.py]: I fixed that, please review

File: mlperf_logging/result_summarizer/result_summarizer.py
Comment (azrael417) [reply mlperf_logging/result_summarizer/result_summarizer.py]: fixed

File: mlperf_logging/result_summarizer/result_summarizer.py
Comment (azrael417) [reply mlperf_logging/result_summarizer/result_summarizer.py:296]: fixed

**Comment by shangw-nvidia** [comment]
@emizan76 I created a PR to this PR's source branch (https://github.com/azrael417/logging/pull/3) "FileNotFoundError" that you saw. The root cause was that the paths in `enqueue_config` inside rule yaml are not updated to reflect the split between training and HPC. @azrael417 once you merge https://github.com/azrael417/logging/pull/3, @emizan76 and I can validate this branch again.

**Comment by azrael417** [comment]
Hi all,

the error you mentioned was fixed in my updated PR already. Can you please re verify that the commands work? Btw, the verify training script as-is does not work, because it does not pass `--rcp_bert_train_samples` to the package checker, thus causing the RCP checks to fail. If you run the package checker manually with that flag specified, it works fine, same for result summarizer. Pleaser verify the latest push.

File: mlperf_logging/rcp_checker/rcp_checker.py
Comment (azrael417) [reply mlperf_logging/rcp_checker/rcp_checker.py:373]: fixed

**Comment by azrael417** [comment]
Let me summarize what I found running the tests:

1) fails on Google and NVIDIA submissions because `--rcp_bert_train_samples` is not passed to package checker in that script. But this is an intended fail, not a bug in the scripts.

2) python3 -m mlperf_logging.package_checker training_results_v1.0/Google/ training 1.0.0 --rcp_bert_train_samples works fine, w/o the last flag it fails Rcp test

3) Works fine

4) Works fine

5) Works fine

6) Both work fine

Let me know what you think.

**Comment by shangw-nvidia** [comment]
> Hi all,
> 
> the error you mentioned was fixed in my updated PR already. Can you please re verify that the commands work? Btw, the verify training script as-is does not work, because it does not pass `--rcp_bert_train_samples` to the package checker, thus causing the RCP checks to fail. If you run the package checker manually with that flag specified, it works fine, same for result summarizer. Pleaser verify the latest push.

The `FileNotFoundError` are from places like  https://github.com/mlcommons/logging/blob/5f08a416ab4cff5035063f7b578eaa3dc62af002/mlperf_logging/compliance_checker/training_1.0.0/closed_resnet.yaml#L11 where it should be ` enqueue_config('training_1.0.0/closed_resnet_{}.yaml'.format(v['value']))` instead because you renamed the directories of those rules. Checking the `Files changed` list of this PR as of now, they don't seem to be fixed.

`--rcp_bert_train_samples` is not needed for Google's v1.0 results since their results were reporting number of samples in the "right" place.

File: mlperf_logging/package_checker/package_checker.py
Comment (azrael417) [reply mlperf_logging/package_checker/package_checker.py]: fixed

File: mlperf_logging/compliance_checker/mlp_compliance.py
Comment (azrael417) [reply mlperf_logging/compliance_checker/mlp_compliance.py:44]: done, works

File: mlperf_logging/package_checker/package_checker.py
Comment (shangw-nvidia) [reply mlperf_logging/package_checker/package_checker.py]: Ah good catch! I didn't notice the assertion style inside other existing code.

File: mlperf_logging/package_checker/package_checker.py
Comment (shangw-nvidia) [reply mlperf_logging/package_checker/package_checker.py]: I like your refactor

**Comment by azrael417** [comment]
Hi guys, any update on this?

**Review by shangw-nvidia — APPROVED** [review]

**Comment by shangw-nvidia** [comment]
@xyhuang @emizan76 we'd like to merge this PR now then add in HPC specific things in later PRs.

**Comment by emizan76** [comment]
LGTM

**Review by emizan76 — APPROVED** [review]

**Comment by azrael417** [comment]
Thanks everybody for the quick feedback and help. Who is able to authorize the merge?

**Comment by shangw-nvidia** [comment]
> Thanks everybody for the quick feedback and help. Who is able to authorize the merge?

@xyhuang has the authorization to merge. Although the weird thing is that I'm seeing "At least 1 approving review is required by reviewers with write access." on this webpage. @emizan76 I thought you and I have write access before?

**Comment by azrael417** [comment]
Any update on this? It would be good if we can merge this soon?

**Review by johntran-nv — APPROVED** [review]
approving given other parties approved

**Comment by johntran-nv** [comment]
looks like my approval didn't help either - i must not have write access either, so we need someone who has write access.  @mwawrzos , do you have write access as well?

**Review by xyhuang — APPROVED** [review]

**Pull request merged**
